Spam regulation and Kick command & fixed solution to use /externals/craft.net not ../craft.net

tuomas56 opened the pull request:
Spam system:
you can only send a message 4 times in a row,
counter is reset when you send a different message

Kick command
only ops can kick
when you are kicked it shows "You were kicked by [player]" ([player] being the person who kicked you)
it also prints "[player] was kicked by [player]" to the server chat.

I have also fixed compile order and project dependencies location so that it will compile if you only fork PartyCraft.

tuomas56 committed:
Spam controller added and Kick command

kick limited to ops, if you say something four times in a row, you will
not be able to say that message again until you say something else.
I also fixed compile order and external project location (would not
compile unless you had also forked Craft.Net)

ddevault commented on PartyCraft/Commands/KickCommand.cs:
```diff
@@ -0,0 +1,42 @@
+            AllowedGroups = new List<string>(new[] { "server.op" });
+        }
+
+        
```
Too much whitespace here.

ddevault commented on PartyCraft/Commands/KickCommand.cs:
```diff
@@ -0,0 +1,42 @@
+            catch
+            {
+                user.SendChat("Incorrect Usage.\n" + Documentation);
+            }
```
Check the usage conditions properly, don't just throw it in a try...catch

ddevault commented on PartyCraft/SpamController.cs:
```diff
@@ -0,0 +1,50 @@
+    class SpamController
+    {
+        public static Dictionary<string, int> sd = new Dictionary<string, int>();
+        public static Dictionary<string, string> pm = new Dictionary<string, string>();
```
Use better variable names. Also, don't make this static, make it specific to each instance of Server.

ddevault commented on PartyCraft/SpamController.cs:
```diff
@@ -0,0 +1,50 @@
+
+namespace PartyCraft
+{
+    class SpamController
```
Explicitly declare this `internal`

ddevault commented on PartyCraft/SpamController.cs:
```diff
@@ -0,0 +1,50 @@
+        public static void Init(string name)
+        {
+                sd.Add(name, 0);
+                pm.Add(name, "");
```
Use `string.Empty`

tuomas56 replied:
Ok, just making corrections.
Thanks very much!

On Sun, Jun 2, 2013 at 4:46 PM, Drew DeVault notifications@github.comwrote:

> In PartyCraft/SpamController.cs:
> 
> > +using System.Text;
> > +using Craft.Net;
> > +using Craft.Net.Data;
> > +using Craft.Net.Server;
> > +
> > +namespace PartyCraft
> > +{
> > -    class SpamController
> > -    {
> > -        public static Dictionary<string, int> sd = new Dictionary<string, int>();
> > -        public static Dictionary<string, string> pm = new Dictionary<string, string>();
> >   +
> > -        public static void Init(string name)
> > -        {
> > -                sd.Add(name, 0);
> > -                pm.Add(name, "");
> 
> Use string.Empty
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/SirCmpwn/PartyCraft/pull/13/files#r4491689
> .

tuomas56 replied:
When changing the static dictionaries so they're specific, where would I
set them to a new dictionary object?
is there some kind of place I can put an Initialize() function in the
server script?

On Sun, Jun 2, 2013 at 4:51 PM, Tuomas Laakkonen pigworts2@gmail.comwrote:

> Ok, just making corrections.
> Thanks very much!
> 
> On Sun, Jun 2, 2013 at 4:46 PM, Drew DeVault notifications@github.comwrote:
> 
> > In PartyCraft/SpamController.cs:
> > 
> > > +using System.Text;
> > > +using Craft.Net;
> > > +using Craft.Net.Data;
> > > +using Craft.Net.Server;
> > > +
> > > +namespace PartyCraft
> > > +{
> > > -    class SpamController
> > > -    {
> > > -        public static Dictionary<string, int> sd = new Dictionary<string, int>();
> > > -        public static Dictionary<string, string> pm = new Dictionary<string, string>();
> > >   +
> > > -        public static void Init(string name)
> > > -        {
> > > -                sd.Add(name, 0);
> > > -                pm.Add(name, "");
> > 
> > Use string.Empty
> > 
> > —
> > Reply to this email directly or view it on GitHubhttps://github.com/SirCmpwn/PartyCraft/pull/13/files#r4491689
> > .

ddevault replied:
Just swap it out for a constructor and create the object in the same place everything else is created in Server.cs.

tuomas56 replied:
I have committed changes, shall I submit another pull request?

On Sun, Jun 2, 2013 at 5:04 PM, Drew DeVault notifications@github.comwrote:

> In PartyCraft/SpamController.cs:
> 
> > +using System.Text;
> > +using Craft.Net;
> > +using Craft.Net.Data;
> > +using Craft.Net.Server;
> > +
> > +namespace PartyCraft
> > +{
> > -    class SpamController
> > -    {
> > -        public static Dictionary<string, int> sd = new Dictionary<string, int>();
> > -        public static Dictionary<string, string> pm = new Dictionary<string, string>();
> >   +
> > -        public static void Init(string name)
> > -        {
> > -                sd.Add(name, 0);
> > -                pm.Add(name, "");
> 
> Just swap it out for a constructor and create the object in the same place
> everything else is created in Server.cs.
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/SirCmpwn/PartyCraft/pull/13/files#r4491753
> .

ddevault replied:
Nope, as you commit more changes, the pull request is automatically updated.

tuomas56 replied:
Sorry, missed that. will fix

On Sun, Jun 2, 2013 at 5:18 PM, Drew DeVault notifications@github.comwrote:

> In PartyCraft/SpamController.cs:
> 
> > +using System.Text;
> > +using Craft.Net;
> > +using Craft.Net.Data;
> > +using Craft.Net.Server;
> > +
> > +namespace PartyCraft
> > +{
> > -    class SpamController
> > -    {
> > -        public static Dictionary<string, int> sd = new Dictionary<string, int>();
> > -        public static Dictionary<string, string> pm = new Dictionary<string, string>();
> >   +
> > -        public static void Init(string name)
> > -        {
> > -                sd.Add(name, 0);
> > -                pm.Add(name, "");
> 
> Nope, as you commit more changes, the pull request is automatically
> updated.
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/SirCmpwn/PartyCraft/pull/13/files#r4491793
> .

tuomas56 committed:
updated KickCommand, Server and SpamController

SpamController class now needs object reference, removed whitespace from
KickCommand.cs and create spamController object in Server.cs

tuomas56 committed:
changed spamController to internal.

changed spamController to internal so it doesn't give me errors about
accessibility and updated names for spamController.Init() on login and
spamController.Remove() on logout.

ddevault commented on PartyCraft.sln:
```diff
@@ -1,25 +1,42 @@
 ﻿
 Microsoft Visual Studio Solution File, Format Version 12.00
-# Visual Studio 2012
+# Visual Studio Express 2012 for Windows Desktop
```
Change this back, please. Use MonoDevelop or SharpDevelop if you can't get Visual Studio.

ddevault replied:
This remains a problem.

tuomas56 replied:
Have fixed. sorry for taking so long with the other things.

On Tue, Jul 16, 2013 at 4:04 PM, Drew DeVault notifications@github.comwrote:

> In PartyCraft.sln:
> 
> > @@ -1,25 +1,42 @@
> > 
> >  Microsoft Visual Studio Solution File, Format Version 12.00
> > -# Visual Studio 2012
> > +# Visual Studio Express 2012 for Windows Desktop
> 
> This remains a problem.
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/SirCmpwn/PartyCraft/pull/13/files#r5217878
> .

ddevault commented on PartyCraft/Commands/KickCommand.cs:
```diff
@@ -0,0 +1,42 @@
+            if (parameters.Length == 1)
+            {
+                server.MinecraftServer.SendChat(parameters[0] + " was kicked by " + user.Username);
+                server.MinecraftServer.GetClient(parameters[0]).SendPacket(new DisconnectPacket("You were kicked by " + user.Username));
```
I think there's a MinecraftClient.Disconnect method. Use that. Also, check to see if the specified client is online first, instead of throwing an exception. If you throw an exception, the user of the command is disconnected.

ddevault commented:
Take out the spam protection, I think that'd be better as a plugin.

tuomas56 commented:
Ok. Just editing.

tuomas56 committed:
Delete SpamController.cs

tuomas56 committed:
Update KickCommand.cs

tuomas56 commented:
Have taken out spamProtection and edited kickCommand

tuomas56 committed:
Update PartyCraft.sln

ddevault commented:
I'll review and merge this later today.

tuomas56 commented:
Thank you.

On Tue, Jul 16, 2013 at 5:05 PM, Drew DeVault notifications@github.comwrote:

> I'll review and merge this later today.
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/SirCmpwn/PartyCraft/pull/13#issuecomment-21052699
> .

ddevault commented:
Sorry this has been out there for so long. I need to do some serious reworking of PartyCraft to get it to fit well into the latest changes to Craft.Net, and I'll make sure this is included.
